Makefile

```diff
@@ -266,9 +266,10 @@ ifeq ($(shell command -v flake8 2> /dev/null),)
 endif
 	golangci-lint run --timeout 5m
 	flake8 test/deps/import-busybox
-	shellcheck --shell sh test/*.sh test/includes/*.sh test/suites/*.sh test/backends/*.sh test/lint/*.sh
+	shellcheck --shell sh $(shell git ls-files --cached --modified --others '*.sh' | grep -v '^test/extras/')
```

Comment:
Does this only work with committed files? If so this is no good as we should be able to run static analysis checks before we commit. If so can you add a comment here in the makefike for future readers as I seem to recall making this point in the past to someone.

Reply:
It does work with modified and uncommitted files:

```
$ echo >> test/suites/warnings.sh
$ touch test/suites/sdeziel-warnings.sh
$ git ls-files --cached --modified --others '*.sh' | grep -v '^test/extras/' | grep warnings
test/suites/sdeziel-warnings.sh
test/suites/warnings.sh
test/suites/warnings.sh
```

The caveat is that modified files do show up twice. This could be papered over with `sort -u` but I'll follow your advise and go back to using `find`.